src/seat.c

```diff
@@ -143,6 +143,19 @@ configure_libinput(struct wlr_input_device *wlr_input_device)
 		wlr_log(WLR_INFO, "dwt configured");
 		libinput_device_config_dwt_set_enabled(libinput_dev, dc->dwt);
 	}
+
+	if (libinput_device_config_send_events_get_modes(libinput_dev) !=
```

Comment:
I think this should be a bitwise operation, so something like

```c
uint32_t possible_modes = libinput_device_config_send_events_get_modes(libinput_dev);
if (possible_modes & LIBINPUT_CONFIG_SEND_EVENTS_DISABLED_ON_EXTERNAL_MOUSE) {
    fprintf(stderr, "info: we support send-events-disabled-on-external-mouse\n");
}
```

Reply:
I personally do not understand why a bitwise operation may be better in this case. From what I understand the two implementations serve slightly different purposes and which one is serves our purpose may depend on the specific requirements of our program.

My implementation is trying to set the send events mode to `LIBINPUT_CONFIG_SEND_EVENTS_DISABLED_ON_EXTERNAL_MOUSE` if the current mode is not `LIBINPUT_CONFIG_SEND_EVENTS_DISABLED`. It checks if the operation is successful and logs the result. This is a direct approach to set a specific mode and handle the result.

Your suggestion from my understanding, on the other hand, is checking if the device supports the `LIBINPUT_CONFIG_SEND_EVENTS_DISABLED_ON_EXTERNAL_MOUSE` mode. It uses bitwise AND operation to check if the mode is present in the possible modes. This is a good way to check for support of a specific mode, but it doesn't actually set the mode.

I am still a bit muddled with the requirements. So, if our goal is to set a specific mode and handle the result, my implementation should work well. If we just want to check if a specific mode is supported, the second implementation is more suitable.

However, in a more comprehensive implementation, I suggest a combination of the two approaches, as in we might want to first check if a mode is supported (like in your implementation) and then try to set it (like in my implementation). This would ensure that we only try to set modes that are actually supported by the device, which could help to avoid unnecessary errors.

Please do correct me if you find some error in my understanding:)

Reply:
I'd suggest something like this instead:
```c
if (dc->send_events_mode >= LIBINPUT_CONFIG_SEND_EVENTS_ENABLED) {
	uint32_t possible_modes = libinput_device_config_send_events_get_modes(libinput_dev);
	if (dc-send_events_mode == LIBINPUT_CONFIG_SEND_EVENTS_ENABLED
			|| possible_modes & dc->send_events_mode) {
		libinput_device_config_send_events_set_mode(libinput_dev, dc->send_events_mode);
	}
}
```
With `dc->send_events_mode` defaulting to `-1`.
The reason for the weird `== LIBINPUT_CONFIG_SEND_EVENTS_ENABLED || possible_modes & mode` check is that libinput uses `0` for the enabled mode which obviously can't be checked within a bitset and is always supported: https://gitlab.freedesktop.org/libinput/libinput/-/blob/main/src/libinput.h#L5012

Kudos to malm for finding the enum definition.

Reply:
Thanks for the suggestions, I am going to push a commit that will contain your suggestions along with @johanmalm 's approach of 
> ```
> if (l->send_events_mode < 0) {
>     /* let's not configure this because the user didn't ask us to */
> } else {
>     /* do some stuff with libinput_device_config_send_events_set_mode() */
> }
> ```
putting your suggestions in the else block along with relevant error and info messages.